Orange/widgets/data/owfeatureconstructor.py

```diff
@@ -237,21 +269,54 @@ def modified(self):
     modified = Property(bool, modified, setModified,
                         notify=modifiedChanged)
 
-    def setEditorData(self, data):
+    def setEditorData(self, data, domain):
         self.nameedit.setText(data.name)
-        self.expressionedit.setPlainText(data.expression)
+        self.expressionedit.setText(data.expression)
         self.setModified(False)
         self.featureChanged.emit()
+        self.attrs_model[:] = ["Select feature"]
+        if domain:
+            self.attrs_model[:] += chain(domain.attributes,
+                                         domain.class_vars,
+                                         domain.metas)
 
     def editorData(self):
         return FeatureDescriptor(name=self.nameedit.text(),
-                                 expression=self.nameedit.toPlainText())
+                                 expression=self.nameedit.text())
 
     def _invalidate(self):
         self.setModified(True)
         self.featureEdited.emit()
         self.featureChanged.emit()
 
+    def on_attrs_changed(self):
+        index = self.attributescb.currentIndex()
+        if index > 0:
+            attr = sanitized_name(self.attrs_model[index].name)
+            self.insert_into_expression(attr)
+            self.attributescb.setCurrentIndex(0)
+
+    def on_funcs_changed(self):
+        index = self.functionscb.currentIndex()
+        if index > 0:
+            func = self.funcs_model[index]
+            if func in ["atan2", "fmod", "ldexp", "log", "pow"]:
```

Comment:
I am not a big fan of these "enumeration" if checks. It is easy to miss something (e.g. copysign and hypot also have 2 arguments) and they tend not to be updated.
But even more relevantly here, some functions can be used with 1 or 2 arguments (e.g. log).

I propose a simpler distinction:
if callable(func): add parentheses